dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java

```diff
@@ -62,16 +65,53 @@ private enum Format {
       "timestamp > CURRENT_TIMESTAMP(0) - INTERVAL '14 days'";
 
   ImmutableList<AssessmentQuery> generateAssessmentQueries() {
-    return ImmutableList.of(
-        AssessmentQuery.createMetricsSelect(Format.TABLE_STORAGE_METRICS, UPPER_UNDERSCORE),
-        AssessmentQuery.createShow("WAREHOUSES", Format.WAREHOUSES, LOWER_UNDERSCORE),
-        SHOW_EXTERNAL_TABLES,
-        AssessmentQuery.createShow("FUNCTIONS", Format.FUNCTION_INFO, LOWER_UNDERSCORE));
+    return generateAssessmentQueries(null);
+  }
+
+  ImmutableList<AssessmentQuery> generateAssessmentQueries(@Nullable ConnectorArguments arguments) {
+    ImmutableList.Builder<AssessmentQuery> builder = ImmutableList.builder();
+    
+    // Add database filtering to assessment queries if limit-to-databases is provided
+    String databaseFilter = "";
+    if (arguments != null && !arguments.getLimitToDatabases().isEmpty()) {
+      String quotedNames = arguments.getLimitToDatabases().stream()
+          .map(SnowflakeMetadataConnector::databaseNameQuoted)
+          .collect(Collectors.joining(", "));
+      databaseFilter = String.format(" IN DATABASE %s", quotedNames);
+    }
+    
+    builder.add(AssessmentQuery.createMetricsSelect(Format.TABLE_STORAGE_METRICS, UPPER_UNDERSCORE));
+    builder.add(AssessmentQuery.createShow("WAREHOUSES", Format.WAREHOUSES, LOWER_UNDERSCORE));
+    
+    // Add database filter to external tables query if provided
+    if (!databaseFilter.isEmpty()) {
+      String externalTablesQuery = String.format("SHOW EXTERNAL TABLES%s", databaseFilter);
+      builder.add(AssessmentQuery.createCustomShow(externalTablesQuery, Format.EXTERNAL_TABLES, LOWER_UNDERSCORE));
+    } else {
+      builder.add(SHOW_EXTERNAL_TABLES);
+    }
+    
+    builder.add(AssessmentQuery.createShow("FUNCTIONS", Format.FUNCTION_INFO, LOWER_UNDERSCORE));
+    
+    return builder.build();
   }
 
   ImmutableList<Task<?>> generateLiteSpecificQueries() {
+    return generateLiteSpecificQueries(null);
+  }
+
+  ImmutableList<Task<?>> generateLiteSpecificQueries(@Nullable ConnectorArguments arguments) {
     String view = "SNOWFLAKE.ACCOUNT_USAGE";
     String filter = " WHERE DELETED IS NULL";
+    
+    // Add database filtering if limit-to-databases is provided
```

Comment:
This seems like yet another instance of the pattern repeated in the other file (although with a string concatenation instead of a builder). Maybe you can just create a helper function that return Optional<String> that can be applied to all these instances.